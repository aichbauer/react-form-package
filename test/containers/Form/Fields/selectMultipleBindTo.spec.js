import React from 'react';
import renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16';
import { configure, mount } from 'enzyme';

import {
  Form,
  Field,
  Select,
} from '../../../../src';

configure({ adapter: new Adapter() });

test('Render Form with select input | bindTo', () => {
  const event = {
    target: {
      id: 'emailTemplate',
      value: 'friends',
    },
  };
  const rules = {
    type: 'select',
    min: '-1',
    max: '-1',
    required: false,
  };

  const myComponent = (
    <Form>
      <Select
        id="emailTemplate"
        bindTo={[
          'subject',
          'body',
        ]}
        bindToAlways
        bindToCallback={(value) => {
          if (value === 'friends') {
            return [
              'Friends',
              'Friends Friends',
            ];
          }

          return [
            'Boss',
            'Boss Boss',
          ];
        }}
      >
        <option disabled value="">
          --- Select an email template ---
        </option>
        <option value="friends">Friends</option>
        <option value="boss">Boss</option>
      </Select>
      <Field
        id="subject"
        type="text"
      />
      <Field
        id="body"
        type="textarea"
        rows="5"
        cols="40"
      />
    </Form>
  );

  const tree = renderer.create(myComponent).toJSON();
  const wrapper = mount(myComponent);
  let state = wrapper.state();

  expect(state.data.subject.value).toBe('');
  expect(state.data.body.value).toBe('');

  wrapper.find('select#emailTemplate').simulate('change', event, rules);
  wrapper.update();

  state = wrapper.state();

  expect(state.data.subject.value).toBe('Friends');
  expect(state.data.body.value).toBe('Friends Friends');

  expect(tree).toMatchSnapshot();
});
