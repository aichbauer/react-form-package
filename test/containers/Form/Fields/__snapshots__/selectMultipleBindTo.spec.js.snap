// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Form with select input | bindTo 1`] = `
<form>
  <select
    bindTo={
      Array [
        "subject",
        "body",
      ]
    }
    bindToAllways={true}
    bindToCallback={[Function]}
    className="rfp-select"
    id="emailTemplate"
    invalid={false}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    valid={false}
    value=""
  >
    <option
      disabled={true}
      value=""
    >
      --- Select an email template ---
    </option>
    <option
      value="friends"
    >
      Friends
    </option>
    <option
      value="boss"
    >
      Boss
    </option>
  </select>
  <input
    className="rfp-input"
    id="subject"
    invalid={false}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    type="text"
    valid={false}
    value=""
  />
  <textarea
    className="rfp-textarea"
    cols="40"
    id="body"
    invalid={false}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    rows="5"
    type="textarea"
    valid={false}
    value=""
  />
</form>
`;
